src/executables/sysrepo-plugind.c

```diff
@@ -527,6 +534,34 @@ main(int argc, char **argv)
         goto cleanup;
     }
 
+    /* No need to create a pid file if we're in debug mode (we're not a daemon) */
+
+    if (debug && pidfile) {
+        SRPLG_LOG_WRN("sysrepo-plugind", "Ignoring pidfile in debug mode.");
+    }
+
+    if (!debug && pidfile) {
```

Comment:
This and the later new code block could be put into functions to avoid cluttering main with all this.